plutus-contract/src/Plutus/Contract/StateMachine.hs

```diff
@@ -312,10 +321,22 @@ runStep ::
     -- ^ The input to apply to the state machine
     -> Contract w schema e (TransitionResult state input)
 runStep smc input =
-    runGuardedStep smc input (\_ _ _ -> Nothing) >>= pure . \case
+    runGuardedStep smc input (\_ _ _ -> Nothing) <&> \case
         Left a  -> absurd a
         Right a -> a
 
+getThreadToken :: AsSMContractError e => Contract w schema e ThreadToken
+getThreadToken = mapError (review _SMContractError) $ do
+    ownPK <- ownPubKey
+    let constraints = mustPayToPubKey (pubKeyHash ownPK) (Ada.lovelaceValueOf 1)
+    utx <- either (throwing _ConstraintResolutionError) pure (Constraints.mkTx @Void mempty constraints)
+    tx <- balanceTx utx
+    txOutRef <-
```

Comment:
Is this all a fancy way of saying "get me an output belonging to the user, any output"?

Reply:
In particular I had to stare at it a bit because I was confused that the transaction wasn't getting spent :sweat_smile: 

This function probably needs a note saying that it latches on to a currently-unspent UTXO and you need to be careful in case it gets spent between you doing this and forging the token.